templates/cicd/pipeline_cfn.yml

```diff
@@ -234,16 +250,28 @@ Resources:
               - Name: SCCheckoutArtifact
             OutputArtifacts:
               - Name: BuildOutput
-{{$length := len .Stages}}{{if gt $length 0}}{{range $stage := .Stages}}{{$svcNum := len $stage.LocalServices}}{{if gt $svcNum 0}}
+{{$length := len .Stages}}{{if gt $length 0}}{{range $index, $stage := .Stages}}{{$svcNum := len $stage.LocalServices}}{{if gt $svcNum 0}}
         - Name: DeployTo-{{$stage.Name}}
           Actions:{{if $stage.Prod}}
             - Name: ApprovePromotionTo-{{$stage.Name}}
               ActionTypeId:
                 Category: Approval
                 Owner: AWS
-                Provider: Manual
                 Version: 1
+                Provider: Manual
               RunOrder: 1{{end}}{{range $svc := $stage.LocalServices}}
+          {{if $stage.TestCommands}}
```

Comment:
Will this result in an extra new line?

Reply:
Yes-- thanks! Fixed and also moved the svc range down, to after this if block.

Reply:
I also added a `-` (after `{{`) at the front of the line before DeployTo to prevent an extra new line there.